mrp_bom_attribute_match/models/mrp_bom.py

```diff
@@ -124,11 +124,15 @@ def _check_component_attributes(self):
                     )
                 )
             if not all(attr in prod_attrs for attr in comp_attrs):
+                missing_attrs = rec.env["product.attribute"].browse(
+                    [attr.id for attr in set(prod_attrs) - set(comp_attrs)]
+                )
                 raise ValidationError(
                     _(
                         "Some attributes of the dynamic component are not included into "
-                        "production product attributes."
+                        "production product attributes. Missing attributes: %s"
                     )
+                    % ", ".join(missing_attrs.mapped("name"))
```

Comment:
nitpicking : use gettext built-in string interpolation

```suggestion
                        "production product attributes. Missing attributes: %s",
                        ", ".join(missing_attrs.mapped("name"))
                    )
```